### DISCUSSION TITLE
Is it possible to use go functions instead of api rules?

### DISCUSSION THREAD

**Discussion by iskaa02** [pocketbase/pocketbase#4987] [Q&A]
As the title says, the api rules can get complicated quite quickly, and limited as well, cloud we use go functions, this might be trivial for create, update, or delete request you can just add a hook before the action, but what about the list and view request, you can use a hook but when expanding a field from another collection the function wouldn't be invoked, is there any workaround this? or just disable expanding completely?

**Comment by ganigeorgiev** [comment]
No, there is no such functionality at the moment.

> ...the api rules can get complicated quite quickly, and limited as well...

It would help to elaborate what is your use case and what are the mentioned limitations to eventually fix them in the future.

**Comment by iskaa02** [comment]
Some rules i use have multiple checks that are harder to read and continue to be more unreadable as the projects grow, this is not a big deal, but since i'm already using pocketbase as a framework i think having the option to use go functions instead of API rules makes sense.

Also rules could be limited when you want to check something external, For example: i use another sqlite database (different from the main) as a cache db, i cannot access this db in rules.

I think API rule is a nice feature for simpler checks but when things get complicated having the power of a programming language is important.
adding more features to rules to cover more use cases doesn't solve the issue and it just make rules more complicated.
